sklearn/decomposition/_dict_learning.py

```diff
@@ -1142,6 +1142,33 @@ def transform(self, X):
         check_is_fitted(self)
         return self._transform(X, self.components_)
 
+    def _inverse_transform(self, code, dictionary):
+        """Private method allowing to accommodate both DictionaryLearning and
+        SparseCoder."""
+        if self.split_sign:
+            n_samples, n_features = code.shape
+            n_features //= 2
+            code = code[:, :n_features] - code[:, n_features:]
+
+        return code @ dictionary
+
+    def inverse_transform(self, X):
+        """Transform data back to its original space.
+
+        Parameters
+        ----------
+        X : ndarray of shape (n_samples, n_components)
+            Data to be transformed back. Must have the same number of
+            components as the data used to train the model.
+
+        Returns
+        -------
+        X_new : ndarray of shape (n_samples, n_features)
+            Transformed data.
+        """
+        check_is_fitted(self)
+        return self._inverse_transform(X, self.components_)
```

Comment:
Why going through an additional private function ? The code is short enough and there's barely nothing else besides directly calling this private function here.

Reply:
good point. I did exactly the same as for the tranform method implemented just above to keep the same logic but I agree that it is not necessary. Tell me what you prefer and I will stick to it (and maybe change the transform method too?)

Reply:
Oh I see, it's because `SparseCoder` is stateless so it was done to avoid calling `check_is_fitted` because fit is not mandatory.

It also means that `inverse_transform` doesn't make sense for SparseCoder because since it's stateless, it has no `n_components_`. I think we should restrict to (MiniBatch)DictionaryLearning. In that case defining a private _transform in `_BaseSparseCoding` and public transforms in `DictionaryLearning` and `MiniBatchDictionaryLearning` would be the simplest solution imo.

Reply:
For me it makes a lot of sens that SparseCoder also can inverse_transform (even in the doc they reconstruct the original signal with  `x = np.ravel(np.dot(x, D))` in [there](https://scikit-learn.org/1.5/auto_examples/decomposition/plot_sparse_coding.html#sphx-glr-auto-examples-decomposition-plot-sparse-coding-py) which could be done with inverse transform instead. Maybe we can define   a public `inverse_transform` in  `_BaseSparseCoding`but overwrite the method on SparseCoding to avoid the is_fitted instead?

Reply:
Yes I spoke too quickly. Actually `dictionary` is given as a parameter to SparseCoder, so its inverse_transform is `code@self.dictionary`. So I agree we can still define `SparseCoder.inverse_transform`.

Reply:
OK will do that then. I had no tests on SparseCoder it would have failed because of t fitted test but will add it now.

Reply:
It would also have failed because there's no `n_components_`. So we need to do the same thing as for `transform`, i.e. override it in SparseCoder.